### PR TITLE
porting: finalize: add instructions for official UBports support

### PR DESCRIPTION
**Pull request by amartinz**
Requirements for a port to get officially supported by UBports.

### CODE REVIEW

**Comment by amartinz** [comment]
I am happy with the initial state, however there are some TODOs left and of course a proper review is needed as well :)

**Comment by amartinz** [comment]
Add "MUST be able to install OTAs" to requirements

**Comment by amartinz** [comment]
- Changed emergency calling support from `MUST` to `SHOULD`, as not everyone has access to verifying such functionality.

- Added `Installer` and `Recovery` sections.
```
Installer
---------
* Devices MUST support installation via `UBports Installer`_.

Recovery
--------

* Devices MUST support resetting to factory defaults.
* Devices MUST support updating via OTA.
```

File: porting/finalize/UBports-official-support/index.rst
Comment (amartinz) [inline porting/finalize/UBports-official-support/index.rst:8]: Fill this out

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (amartinz) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:19]: Fill this out

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (amartinz) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:21]: Fill this out

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (amartinz) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:93]: Verify this and remove once verified

**Review by amartinz — COMMENTED** [review]
Marked TODOs for easier visibility

**Comment by amartinz** [comment]
- Added `Documentation` section to requirements:
```
Documentation
-------------

* The device MUST be listed at the `Ubuntu Touch device page`_; the device page needs to be kept up-to-date.
```

- Added workflow for `Submitting your port` section, utilizing our new `devrel` team.
  - Including a brand new port submission template
- Addressed all leftover TODOs

**Comment by amartinz** [comment]
Should be ready for review now.

I'd like this to get all approvals from the reviewers listed (at least, more are always welcome) before merging this. :smile:

File: porting/finalize/index.rst
Comment (doniks) [inline porting/finalize/index.rst:16]: I would suggest to drop this note. Or at least don't format it as a note. It stands out and interrupts the reading flow, but I don't think this is needed here

File: porting/finalize/UBports-official-support/index.rst
Comment (doniks) [inline porting/finalize/UBports-official-support/index.rst:50]: ...got pull requests ready.... I don't understand

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:13]: Pls move the Attaining exceptions section down so it is just before the current "Exception"  section. 

I think documents should always start to explain the 'normal' stuff first, so a new reader has an easier time grasping what we're talking about here.

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:11]: I don't think we use horizontal lines anywhere else in our docs, pls remove

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:20]: "this" repo? which one? the docs?

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:34]: please spell out RIL. I have no clue what this is. If you think it's clear once it is spelled out, great, otherwise consider explaining/linking an explanation

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:77]: Is this a specific issue? I don't really understand. Can we link smt here?

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:86]: consider using plural "Exceptions"

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:89]: I don't really understand where this list is supposed to go. Where is "this" repository?

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [inline porting/finalize/UBports-official-support/port-support-requirements.rst:98]: we don't have specific licensing agreements on individual docs pages anywhere else. do we really need this here? it seems to me everything we're saying is sufficiently original content/fair use, we don't need this.

**Review by doniks — CHANGES_REQUESTED** [review]

**Comment by doniks** [comment]
sorry, took me a while to get around to this. I'm very glad that we are moving forward in this area. I only have small mostly formatting/formulation remarks. 

However, note that I only review clarity of language and presentation. I don't offer an opinion on the technical side. I think this should come from core developers, so I'd very much prefer another approval.

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (amartinz) [reply porting/finalize/UBports-official-support/port-support-requirements.rst:77]: This is (was?) known in the android community as a device, which is powered on but not responding to any input, like pressing power keys.

We can drop this and readd more points later on i think?

File: porting/finalize/UBports-official-support/index.rst
Comment (amartinz) [reply porting/finalize/UBports-official-support/index.rst]: I have adjusted it, i hope it is more clear now, sorry!

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (amartinz) [reply porting/finalize/UBports-official-support/port-support-requirements.rst]: Yes, the docs one. I have added links to it now to make it more clear.

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (amartinz) [reply porting/finalize/UBports-official-support/port-support-requirements.rst]: I have spelled it out, however i doubt explaining it further helps, unless we got a specific RIL section in our documentation already, which we want to link.

Otherwise the requirements get too verbose.

Comment (doniks) [reply porting/finalize/UBports-official-support/port-support-requirements.rst]: agree. no point in elaborating more *here*. thanks for spelling it out. it at least gives a little more context for an internet search

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [reply porting/finalize/UBports-official-support/port-support-requirements.rst]: ah. ok, yeah. Even after having attempted a port myself, I'm very detached from the android/xda lingo ... And I generally always found it too vague. I think it is better to keep this language out of the docs only only use it with air quotes and amended with technical descriptions. I think in this case it is fine to drop this ... or if you want to you add some more high level .... "It should turn on and actually allow me to see stuff on the screen, start apps and such " :D ... just kidding. I think it's fine as is now

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (amartinz) [reply porting/finalize/UBports-official-support/port-support-requirements.rst:23]: Thanks, forgot to adjust that 🤦

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (amartinz) [reply porting/finalize/UBports-official-support/port-support-requirements.rst:81]: I have adapted this from the Lineage charter and they got an own wiki for devices.
For reference, this is the document i based it on: https://github.com/LineageOS/charter/blob/master/device-support-requirements.md

So my adjustments for UT might be a bit weird and I never got feedback so far for anything, so this is very much appreciated, thank you :)

Comment (doniks) [reply porting/finalize/UBports-official-support/port-support-requirements.rst:81]: ah. I see. Well, if this is not really set in stone, then may I propose to document the exceptions in ONE place only. and make the documentation MUST, so overall how about this:

Exceptions
***********

Device or software deviations from our core product (as described above) MUST be approved by the devrel team.

Process
----

* Exceptions MUST be proposed via a change request to the list of Granted exceptions below in the `docs.ubports.com repository`_.
* Exceptions MUST be approved by the devrel team.

Granted exceptions
------


====  ==========  ================  ======
Date  Device/OEM  Section exempted  Reason
====  ==========  ================  ======
====  ==========  ================  ======

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [reply porting/finalize/UBports-official-support/port-support-requirements.rst:23]: just to avoid us waiting on each other ... you didn't commit anything about this yet

File: porting/finalize/UBports-official-support/port-support-requirements.rst
Comment (doniks) [reply porting/finalize/UBports-official-support/port-support-requirements.rst:81]: @amartinz ping

**Comment by JamiKettunen** [comment]
Was this closed in mistake or?

**Comment by amartinz** [comment]
> Was this closed in mistake or?

I do not have the time to further pursue this MR and do not want to be the blocker in case someone wants to pick this up :)

**Comment by doniks** [comment]
> > Was this closed in mistake or?
> 
> I do not have the time to further pursue this MR and do not want to be the blocker in case someone wants to pick this up :)

I could pick up the leg work of *writing*, but I would need a core dev that reviews the *content* . Otherwise I just make up a process that only exists on paper. @amartinz any idea whom I might contact from the core team?

**Comment by amartinz** [comment]
> I could pick up the leg work of _writing_, but I would need a core dev that reviews the _content_ . Otherwise I just make up a process that only exists on paper. @amartinz any idea whom I might contact from the core team?

The reviewers added to this MR, i guess, as i think in this case it is more important to get feedback and input from the porting team rather the platform team.

**Comment by doniks** [comment]
trying again over here: https://github.com/ubports/docs.ubports.com/pull/540

**Comment by Flohack74** [comment]
@doniks the problem is: Even if only a handful of people need to agree on this I do not want to push this entirely on my own and with Alex. I would have merged this already, but there is little interest in the porting corner to get this moving I guess. It is also unclear who are the stakeholders, and how we empower them to execute all of this.

If we get more restrictive then we should really be serious. It needs to be a logical step forward, but sometimes I have the feeling it is not what people want when saying "mobile freedom". Or idk.

**Pull request closed without merging**
